src/library-authoring/generic/status-widget/index.tsx

```diff
@@ -167,6 +175,25 @@ const StatusWidget = ({
     }
   }
 
+  if (publishButtonState !== 'pending') {
+    if (isPublished && publishButtonState !== 'complete') {
+      setPublishButtonState('complete');
+    } else if (!isPublished && publishButtonState !== 'default') {
+      setPublishButtonState('default');
+    }
+  }
+
+  const handleOnClickPublish = useCallback(() => {
+    if (onCommit) {
+      setPublishButtonState('pending');
+      onCommit().finally(() => {
+        // Set to default to let new 'isPublished' value to decide
+        // the new state of the button.
+        setPublishButtonState('deafult');
```

Comment:
```suggestion
        setPublishButtonState('default');
```
Maybe set a type like `'default' | 'pending' | 'complete'` for the state instead of just `string` to catch this sort of error?

Also, would it be possible and simpler to pass in `commitLibraryChanges.status` to this component, instead of having it track the status in a new state variable?

Reply:
Done: https://github.com/openedx/frontend-app-authoring/pull/2237/commits/ec7fbce70c763820aa449ec7d3116387bc03eef4

Thanks for the recommendation!